lib/core.ps1

```diff
@@ -662,10 +662,26 @@ function Invoke-ExternalCommand {
     return $true
 }
 
-function env($name,$global,$val='__get') {
-    $target = 'User'; if($global) {$target = 'Machine'}
-    if($val -eq '__get') { [environment]::getEnvironmentVariable($name,$target) }
-    else { [environment]::setEnvironmentVariable($name,$val,$target) }
+function env($name, $global, $val = '__get') {
+    $RegisterKey = if ($global) {
+        Get-Item -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager'
+    } else {
+        Get-Item -Path 'HKCU:'
+    }
+
+    if ($val -eq '__get') {
+        $EnvRegisterKey = $RegisterKey.OpenSubKey('Environment')
+        $RegistryValueOption = [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames
+        $EnvRegisterKey.GetValue($name, $null, $RegistryValueOption)
+    } else {
+        $EnvRegisterKey = $RegisterKey.OpenSubKey('Environment', $true)
+        $RegistryValueKind = if ($EnvRegisterKey.GetValue($name)) {
```

Comment:
The String type should be chosen by the existence of `%` in the value, not based on the old value.

Reply:
I don't think so, maybe the user just wants to have `%%` (not expanded) in the environment variable, it should keep the original type in the registry.

It's hard to say what standard we should specify to set the type, maybe keeping it as it is is a feasible way. However, there is indeed a problem here. If the user sets the Path to `REG_SZ` in some cases, there will be problems directly adding `%SCOOP_*%` to the Path.

Reply:
If you add a `%%` variable via System Properties then it will change it automatically to `REG_EXPAND_SZ`. We should mimic the default windows behavior in this case.

Reply:
This is a nice perspective. However, I think it is better to add a parameter to distinguish whether it should be changed. The detection % is too difficult to say that it will definitely meet expectations. It is up to the function caller to decide the new type, otherwise it will remain the same, but this will require modifying the caller in various places, which will involve many places.

I just want to make it simpler, that is, to change the original default behavior of `env` in the smallest scope.

Reply:
I modified my code, you see if it fits.